test/test_va_api_config_attribs.cpp

```diff
@@ -39,40 +40,106 @@ class VAAPIConfigAttribs
+                EXPECT_EQ(sAttrib.value & aAttrib.value, aAttrib.value);
+                break;
+
+            // The following type's return a bitfield of supported choices
+            // where only one choice is allowed in the actual config.
+            case VAConfigAttribRateControl:
+                EXPECT_TRUE(
+                    std::find(g_vaRateControls.begin(), g_vaRateControls.end(),
+                        aAttrib.value) != g_vaRateControls.end())
+                    << "0x" << std::hex << aAttrib.value << std::dec
+                    << " rate control not found.  Only one allowed.";
+                EXPECT_EQ(sAttrib.value & aAttrib.value, aAttrib.value);
+                break;
+
+            case VAConfigAttribDecSliceMode:
+                EXPECT_TRUE(
+                    std::find(g_vaDecSliceModes.begin(),
+                        g_vaDecSliceModes.end(),
+                        aAttrib.value) != g_vaDecSliceModes.end())
+                    << "0x" << std::hex << aAttrib.value << std::dec
+                    << " decode slice mode not found.  Only one allowed";
+                EXPECT_EQ(sAttrib.value & aAttrib.value, aAttrib.value);
+                break;
+
+            // The actual value should match the supported value
+            default:
+                EXPECT_EQ(sAttrib.value, aAttrib.value);
+            }
+        }
+    }
 };
 
-TEST_F(VAAPIConfigAttribs, GetConfigAttribs)
+TEST_P(VAAPIConfigAttribs, GetConfigAttribs)
 {
-    doGetMaxValues();
+    // The driver must support creating a config without any user specified
+    // attributes, in which case the driver will use it's own defaults.  The
+    // app should be able to query those default attributes for such created
+    // config via vaQueryConfigAttributes.  Those default attribute values
+    // should be consistent with supported attribute values returned by
+    // vaGetConfigAttributes.
+
+    const VAProfile& profile        = ::testing::get<0>(GetParam());
+    const VAEntrypoint& entrypoint  = ::testing::get<1>(GetParam());
 
+    doGetMaxValues();
     doQueryConfigProfiles();
-    const Profiles& profiles = getSupportedProfileList();
-    ASSERT_FALSE(profiles.empty());
 
-    for (const auto& profile : profiles) {
+    if (doFindProfileInList(profile)) {
         doQueryConfigEntrypoints(profile);
-        const Entrypoints& entrypoints = getSupportedEntrypointList();
-        ASSERT_FALSE(entrypoints.empty());
+        if (doFindEntrypointInList(entrypoint)) {
+            // create config without attributes (i.e. use driver defaults)
+            createConfig(profile, entrypoint);
 
-        for (const auto& entrypoint : entrypoints) {
-            doGetMaxNumConfigAttribs();
-            doCreateConfigNoAttrib(profile, entrypoint);
+            // query default attributes from the config we just created
+            ConfigAttributes actual;
+            queryConfigAttributes(profile, entrypoint, actual);
```

Comment:
When queryConfigAttributes returns actual.clear() in case of error, proceeding further may not be logical.

Reply:
I disagree.  If it fails, we still need to call `destroyConfig()`... and all other calls after that have proper checks in place to fail appropriately, too (i.e. it won't crash because of empty attributes).  In general, tests should only need to stop proceeding if a failure would cause test to crash in later operations.

Reply:
+1 for "In general, tests should only need to stop proceeding if a failure would cause test to crash in later operations." and "proper checks in place to fail appropriately, too"

In this case Failure will be caught downstream, which allows error propagation rather than catching failure locally. I agree that tests should proceed in non-fatal failures but erroring out locally is probably better for debug.